src/runtime/components/forms/Form.ts

```diff
@@ -0,0 +1,160 @@
+import { provide, ref, type PropType, h, defineComponent } from 'vue'
+import { useEventBus } from '@vueuse/core'
+import type { ZodSchema, ZodError } from 'zod'
+import type { ValidationError as JoiError, Schema as JoiSchema } from 'joi'
+import type { ObjectSchema as YupObjectSchema, ValidationError as YupError } from 'yup'
+import type { FormError, FormEvent } from '../../types'
+
+export default defineComponent({
+  props: {
+    schema: {
+      type: Object as
+        | PropType<ZodSchema>
+        | PropType<YupObjectSchema<any>>
+        | PropType<JoiSchema>,
+      default: undefined
+    },
+    state: {
```

Comment:
Wouldn't it be better to rename `state` to `form`?

Reply:
We could rename the `form` to `formRef` in the examples

Reply:
IMO that would be confusing, having `form` as a reference to the HTML element is a bit more natural. We can rename it to `model` or `data` if you prefer.

Reply:
Let's keep it this way, it was just an idea :)